src/Symfony/Bundle/SecurityBundle/composer.json

```diff
@@ -40,7 +41,6 @@
         "symfony/expression-language": "^5.4|^6.0|^7.0",
         "symfony/form": "^5.4|^6.0|^7.0",
         "symfony/framework-bundle": "^6.4|^7.0",
-        "symfony/http-client": "^5.4|^6.0|^7.0",
```

Comment:
It still can be an option IMO, you would just need to check for it's availability and throw an exception if not found.

Reply:
I replace the require with an exception if HttpClient class was not found.